connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTask.java

```diff
@@ -258,6 +258,21 @@ private void commitTransaction() {
 
         long started = time.milliseconds();
 
+        if (!transactionOpen && !offsetWriter.willFlush()) {
+            // There is no contents on the framework side to commit, so skip the offset flush and producer commit
+            long durationMillis = time.milliseconds() - started;
+            recordCommitSuccess(durationMillis);
+            log.debug("{} Finished commitOffsets successfully in {} ms", this, durationMillis);
+
+            // Synchronize in order to guarantee that writes on other threads are picked up by this one
+            synchronized (commitableRecords) {
+                commitableRecords.forEach(this::commitTaskRecord);
+                commitableRecords.clear();
+            }
```

Comment:
This part shouldn't be necessary, should it? There should be no committable records if there's no open transaction and we don't have any offsets for filtered records to commit.